cmd/mailbox-controller/controller.go

```diff
@@ -252,10 +253,18 @@ func (ctl *mbCtl) sync(ctx context.Context, refany any) bool {
 func (ctl *mbCtl) ensureBinding(ctx context.Context, workspace *tenancyv1alpha1.Workspace) bool {
 	logger := klog.FromContext(ctx).WithValues("mbsName", workspace.Name)
 
-	// TODO verify binding is already in place.
-	// ISSUE kube-bind objects(apiservicebindings) does not have clusterclientset like in KCP. Maybe dynamicClusterClient can help.
+	// The script ensures that a mailbox workspace has the needed KubeStellar CRDs.
+	shellScriptName := "kubectl-kubestellar-ensure-mbw"
 
-	// TODO bind all resources. Note: running in pod, need to access the dex
+	// The script is idempotent.
+	logger.V(2).Info("Executing shell script", "script", shellScriptName)
+	cmdLine := shellScriptName + " " + workspace.Name
+	cmd := exec.Command("/bin/sh", "-c", cmdLine)
+	_, err := cmd.Output()
```

Comment:
The script should provide simple, easy to parse output, in order to be able to map provider namespace for this space and the space name(see TODO in line 269 below).
Something like: "Completed. Service provider namespace is:<ns for this MBW>"

Reply:
Thank you. Exactly, because of this, I didn't touch line 269 in this PR, so it can continue to remind me there is parsing to do here.

I am thinking finishing item 3 in Section "Use kube-bind" of https://github.com/kubestellar/kubestellar/issues/1208 in this PR, and then support "Stop using the cluster-id" Section in the next PR. @andreyod please let me know if we want to also do the parsing here.

Reply:
I think the parsing should be in another PR, together with other changes for "Stop using the cluster-id" that we will need to do after we start using space framework(we are not there yet). But  `kubectl-kubestellar-ensure-mbw` can provide the needed output in this PR.

Reply:
I got the point. But if we go with that way, we will have to do the parsing twice. First we'll parse the stdout/stderr of kubectl-bind in the script, second we'll parse the output of the script in mailbox-controller go code.

I was planning to do the parsing only once and for all, in mailbox-controller go code. What do you think?

Reply:
I think we should postpone that to another PR AFTER we have merged the space framework. The integration would be
1. Create the space through SM
2. Get the space config info from the SM
3. Execute bind using the above config
4. Parse the output to get the NS on the export provider 
5. Set the Space annotation with the info in (4) 
In all the scripts the above will be done by the script (including bring up konnector process) 

In the MB case 1,2 and 3 are done from code. for 4 & 5 we have 2 options:
1. Do (4) & (5)  in code  in order to do (4) the exec need to get all output of the bind, and (5) will be done directly through client-go
2. Have the script do 3, 4 & 5

Reply:
Option 1 makes more sense to me.